src/providers/cryptoauthlib/aead.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+// CAL supports CCM with:
+// - tag lenght must be <4,16> and must be odd number
+// - nonce lenght must be <7,13>
+// CAL supports GCM with:
+// - tag lenght must be <12,16>
+// - nonce lenght must be <7,13>
+
+use super::Provider;
+use crate::authenticators::ApplicationName;
+use crate::key_info_managers::KeyTriple;
+use log::error;
+use parsec_interface::operations::psa_algorithm::{Aead, AeadWithDefaultLengthTag};
+use parsec_interface::operations::{psa_aead_decrypt, psa_aead_encrypt};
+use parsec_interface::requests::{ProviderId, ResponseStatus, Result};
+
+const DEFAULT_TAG_LENGTH: usize = 16;
+
+pub fn get_tag_length(alg: &Aead) -> Option<usize> {
+    match alg {
+        Aead::AeadWithDefaultLengthTag(AeadWithDefaultLengthTag::Ccm) => Some(DEFAULT_TAG_LENGTH),
+        Aead::AeadWithDefaultLengthTag(AeadWithDefaultLengthTag::Gcm) => Some(DEFAULT_TAG_LENGTH),
+        Aead::AeadWithShortenedTag {
+            aead_alg: AeadWithDefaultLengthTag::Ccm,
+            tag_length,
+        } => Some(*tag_length),
+        Aead::AeadWithShortenedTag {
+            aead_alg: AeadWithDefaultLengthTag::Gcm,
+            tag_length,
+        } => Some(*tag_length),
+        _ => None,
+    }
+}
```

Comment:
Reminder to myself that this would be a useful method on [`Aead`](https://docs.rs/psa-crypto/0.9.0/psa_crypto/types/algorithm/enum.Aead.html)